### ISSUE TITLE
LoadError: cannot load such file -- rubygems/util

### ISSUE THREAD

**Issue by ktiniatros** [fastlane/fastlane#12401]
### New Issue Checklist

- [✅ ] Updated fastlane to the latest version
- [ ✅] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [✅ ] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [✅ ] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

My fastlane supply step fails for some reason on my Teamcity Linux agent. (the step is for uploading an android apk to beta). Can someone point me how to debug this?
(Keep in mind I stripped out some irrelevant to the problem code (for the sake of easier debugging) and some personal info from my fastfile)

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>[15:24:55][Step 3/8] LoadError: cannot load such file -- rubygems/util
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/unicode-display_width-1.3.2/lib/unicode/display_width/index.rb:1:in `require'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/unicode-display_width-1.3.2/lib/unicode/display_width/index.rb:1:in `<top (required)>'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/unicode-display_width-1.3.2/lib/unicode/display_width.rb:2:in `require_relative'
[15:24:55][Step 3/8] bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/unicode-display_width-1.3.2/lib/unicode/display_width.rb:2:in `<top (required)>'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/terminal-table-1.8.0/lib/terminal-table/cell.rb:1:in `require'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/terminal-table-1.8.0/lib/terminal-table/cell.rb:1:in `<top (required)>'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/terminal-table-1.8.0/lib/terminal-table.rb:25:in `require'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/terminal-table-1.8.0/lib/terminal-table.rb:25:in `block in <top (required)>'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/terminal-table-1.8.0/lib/terminal-table.rb:24:in `each'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/terminal-table-1.8.0/lib/terminal-table.rb:24:in `<top (required)>'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane/lib/fastlane/lane_manager_base.rb:36:in `require'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane/lib/fastlane/lane_manager_base.rb:36:in `print_table'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane/lib/fastlane/lane_manager_base.rb:18:in `finish_fastlane'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane/lib/fastlane/lane_manager.rb:74:in `cruise_lane'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane/lib/fastlane/commands_generator.rb:107:in `block (2 levels) in run'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane/lib/fastlane/commands_generator.rb:332:in `run'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
[15:24:55][Step 3/8]   /opt/teamcity/.gem/ruby/gems/fastlane-2.93.1/bin/fastlane:20:in `<top (required)>'
[15:24:55][Step 3/8]   /usr/local/bin/fastlane:23:in `load'
[15:24:55][Step 3/8]   /usr/local/bin/fastlane:23:in `<top (required)>'</pre>
</details>

### Environment

<details>
  <pre>[15:47:19][Step 1/8] [15:47:19]: Generating fastlane environment output, this might take a few seconds...
[15:47:19][Step 1/8] [15:47:19]: undefined method `strip' for nil:NilClass
[15:47:19][Step 1/8] [15:47:19]: Error rendering markdown table using the following text:
[15:47:19][Step 1/8] [15:47:19]: | Key | Value |
[15:47:19][Step 1/8] | fastlane | 2.93.1 |
[15:47:19][Step 1/8] | OS | Linux ip-10-0-91-208 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux |
[15:47:19][Step 1/8] | Ruby | 2.0.0 |
[15:47:19][Step 1/8] | Bundler? | true |
[15:47:19][Step 1/8] | Git | git version 1.8.3.1 |
[15:47:19][Step 1/8] | Installation Source | /usr/local/bin/fastlane |
[15:47:19][Step 1/8] | Host | \S
[15:47:19][Step 1/8] Kernel \r on an \m Linux ip-10-0-91-208 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux (3.10.0-514.26.2.el7.x86_64) |
[15:47:19][Step 1/8] | Ruby Lib Dir | /usr/lib64 |
[15:47:19][Step 1/8] | OpenSSL Version | OpenSSL 1.0.1e 11 Feb 2013 |
[15:47:19][Step 1/8] | Is contained | false |
[15:47:19][Step 1/8] | Is homebrew | false |
[15:47:19][Step 1/8] | Is installed via Fabric.app | false |
[15:47:19][Step 1/8] <details><summary>✅ fastlane environment ✅</summary>
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] ### Stack
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] | Key | Value |
[15:47:19][Step 1/8] | fastlane | 2.93.1 |
[15:47:19][Step 1/8] | OS | Linux ip-10-0-91-208 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux |
[15:47:19][Step 1/8] | Ruby | 2.0.0 |
[15:47:19][Step 1/8] | Bundler? | true |
[15:47:19][Step 1/8] | Git | git version 1.8.3.1 |
[15:47:19][Step 1/8] | Installation Source | /usr/local/bin/fastlane |
[15:47:19][Step 1/8] | Host | \S
[15:47:19][Step 1/8] Kernel \r on an \m Linux ip-10-0-91-208 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux (3.10.0-514.26.2.el7.x86_64) |
[15:47:19][Step 1/8] | Ruby Lib Dir | /usr/lib64 |
[15:47:19][Step 1/8] | OpenSSL Version | OpenSSL 1.0.1e 11 Feb 2013 |
[15:47:19][Step 1/8] | Is contained | false |
[15:47:19][Step 1/8] | Is homebrew | false |
[15:47:19][Step 1/8] | Is installed via Fabric.app | false |
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] ### System Locale
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] | Variable | Value       |   |
[15:47:19][Step 1/8] | -------- | ----------- | - |
[15:47:19][Step 1/8] | LANG     | en_US.UTF-8 | ✅ |
[15:47:19][Step 1/8] | LC_ALL   | en_US.UTF-8 | ✅ |
[15:47:19][Step 1/8] | LANGUAGE |             |   |
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] ### fastlane files:
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] <details><summary>`./fastlane/Fastfile`</summary>
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] ```ruby
[15:47:19][Step 1/8] # Customise this file, documentation can be found here:
[15:47:19][Step 1/8] # https://github.com/fastlane/fastlane/tree/master/fastlane/docs
[15:47:19][Step 1/8] # All available actions: https://docs.fastlane.tools/actions
[15:47:19][Step 1/8] # can also be listed using the `fastlane actions` command
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] # If you want to automatically update fastlane if a new version is available:
[15:47:19][Step 1/8] # update_fastlane
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] # This is the minimum version number required.
[15:47:19][Step 1/8] # Update this, if you use features of a newer version
[15:47:19][Step 1/8] fastlane_version "2.63.0"
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] default_platform :android
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] # Supported locales on the Play Store
[15:47:19][Step 1/8] locales = Array["da-DK", "de-DE", "en-GB", "en-US", "nl-NL", "sv-SE"]
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] # Metadata variables
[15:47:19][Step 1/8] metadata_folder = "#{Dir.pwd}/metadata"
[15:47:19][Step 1/8] defaults_folder = [metadata_folder, "defaults"].join("/")
[15:47:19][Step 1/8] version_name_filename = "version_name.txt"
[15:47:19][Step 1/8] whatsnew_filename = "whatsnew.txt"
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] # Play Store distribution tracks
[15:47:19][Step 1/8] beta = "beta"
[15:47:19][Step 1/8] prod = "production"
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] platform :android do
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8]   desc "Builds and uploads a new version to the Play Store, for beta distribution"
[15:47:19][Step 1/8]   lane :deploy_app do |options|
[15:47:19][Step 1/8]     # /!\ package name derived from the flavor name, we need to make sure it always stays like this
[15:47:19][Step 1/8]         supply(package_name: "com.myid." + "myapp",
[15:47:19][Step 1/8]                mapping: "app/build/outputs/mapping/" + "myapp" + "/release/mapping.txt",
[15:47:19][Step 1/8]                metadata_path: [metadata_folder, "myapp"].join("/"),
[15:47:19][Step 1/8]                track: beta,
[15:47:19][Step 1/8]                validate_only: true)
[15:47:19][Step 1/8]   end
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] end
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] # More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
[15:47:19][Step 1/8] # All available actions: https://docs.fastlane.tools/actions
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] # fastlane reports which actions are used. No personal data is recorded.
[15:47:19][Step 1/8] # Learn more at https://github.com/fastlane/fastlane#metrics
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] ```
[15:47:19][Step 1/8] </details>
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] <details><summary>`./fastlane/Appfile`</summary>
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] ```ruby
[15:47:19][Step 1/8] # Path to the json secret file
[15:47:19][Step 1/8] json_key_file ENV["PROJECT_ANDROID_GOOGLE_PLAY_JSON_KEY"]
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] ```
[15:47:19][Step 1/8] </details>
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] ### fastlane gems
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] | Gem      | Version | Update-Status |
[15:47:19][Step 1/8] | -------- | ------- | ------------- |
[15:47:19][Step 1/8] | fastlane | 2.93.1  | ✅ Up-To-Date  |
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] ### Loaded fastlane plugins:
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] **No plugins Loaded**
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] <details><summary><b>Loaded gems</b></summary>
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] | Gem                       | Version      |
[15:47:19][Step 1/8] | ------------------------- | ------------ |
[15:47:19][Step 1/8] | bundler                   | 1.15.1       |
[15:47:19][Step 1/8] | CFPropertyList            | 3.0.0        |
[15:47:19][Step 1/8] | public_suffix             | 2.0.5        |
[15:47:19][Step 1/8] | addressable               | 2.5.2        |
[15:47:19][Step 1/8] | babosa                    | 1.0.2        |
[15:47:19][Step 1/8] | colored                   | 1.2          |
[15:47:19][Step 1/8] | highline                  | 1.7.10       |
[15:47:19][Step 1/8] | commander-fastlane        | 4.4.6        |
[15:47:19][Step 1/8] | dotenv                    | 2.4.0        |
[15:47:19][Step 1/8] | emoji_regex               | 0.1.1        |
[15:47:19][Step 1/8] | excon                     | 0.62.0       |
[15:47:19][Step 1/8] | multipart-post            | 2.0.0        |
[15:47:19][Step 1/8] | faraday                   | 0.15.0       |
[15:47:19][Step 1/8] | unf_ext                   | 0.0.7.5      |
[15:47:19][Step 1/8] | unf                       | 0.1.4        |
[15:47:19][Step 1/8] | domain_name               | 0.5.20180417 |
[15:47:19][Step 1/8] | http-cookie               | 1.0.3        |
[15:47:19][Step 1/8] | faraday-cookie_jar        | 0.0.6        |
[15:47:19][Step 1/8] | faraday_middleware        | 0.12.2       |
[15:47:19][Step 1/8] | fastimage                 | 2.1.1        |
[15:47:19][Step 1/8] | gh_inspector              | 1.1.3        |
[15:47:19][Step 1/8] | jwt                       | 1.5.6        |
[15:47:19][Step 1/8] | little-plugger            | 1.1.4        |
[15:47:19][Step 1/8] | multi_json                | 1.13.1       |
[15:47:19][Step 1/8] | logging                   | 2.2.2        |
[15:47:19][Step 1/8] | memoist                   | 0.16.0       |
[15:47:19][Step 1/8] | os                        | 0.9.6        |
[15:47:19][Step 1/8] | signet                    | 0.8.1        |
[15:47:19][Step 1/8] | googleauth                | 0.6.2        |
[15:47:19][Step 1/8] | httpclient                | 2.8.3        |
[15:47:19][Step 1/8] | mime-types-data           | 3.2016.0521  |
[15:47:19][Step 1/8] | mime-types                | 3.1          |
[15:47:19][Step 1/8] | declarative               | 0.0.10       |
[15:47:19][Step 1/8] | declarative-option        | 0.1.0        |
[15:47:19][Step 1/8] | uber                      | 0.1.0        |
[15:47:19][Step 1/8] | representable             | 3.0.4        |
[15:47:19][Step 1/8] | retriable                 | 3.1.1        |
[15:47:19][Step 1/8] | google-api-client         | 0.13.6       |
[15:47:19][Step 1/8] | json                      | 2.1.0        |
[15:47:19][Step 1/8] | mini_magick               | 4.5.1        |
[15:47:19][Step 1/8] | multi_xml                 | 0.6.0        |
[15:47:19][Step 1/8] | plist                     | 3.4.0        |
[15:47:19][Step 1/8] | rubyzip                   | 1.2.1        |
[15:47:19][Step 1/8] | security                  | 0.1.3        |
[15:47:19][Step 1/8] | naturally                 | 2.1.0        |
[15:47:19][Step 1/8] | simctl                    | 1.6.4        |
[15:47:19][Step 1/8] | slack-notifier            | 2.3.2        |
[15:47:19][Step 1/8] | terminal-notifier         | 1.8.0        |
[15:47:19][Step 1/8] | unicode-display_width     | 1.3.2        |
[15:47:19][Step 1/8] | terminal-table            | 1.8.0        |
[15:47:19][Step 1/8] | tty-screen                | 0.6.4        |
[15:47:19][Step 1/8] | tty-cursor                | 0.5.0        |
[15:47:19][Step 1/8] | tty-spinner               | 0.8.0        |
[15:47:19][Step 1/8] | word_wrap                 | 1.0.0        |
[15:47:19][Step 1/8] | atomos                    | 0.1.2        |
[15:47:19][Step 1/8] | claide                    | 1.0.2        |
[15:47:19][Step 1/8] | colored2                  | 3.1.2        |
[15:47:19][Step 1/8] | nanaimo                   | 0.2.5        |
[15:47:19][Step 1/8] | xcodeproj                 | 1.5.7        |
[15:47:19][Step 1/8] | rouge                     | 2.0.7        |
[15:47:19][Step 1/8] | xcpretty                  | 0.2.8        |
[15:47:19][Step 1/8] | xcpretty-travis-formatter | 1.0.0        |
[15:47:19][Step 1/8] </details>
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] 
[15:47:19][Step 1/8] *generated on:* **2018-04-27**
[15:47:19][Step 1/8] </details></pre>
</details>

**Comment by joshdholtz** [comment]
@ktiniatros Does this fastlane implementation work on your local development machine? And it only fails on your TeamCity Linux agent?

**Comment by ktiniatros** [comment]
Yeah it works locally on my macbook, but it fails on the TeamCity Linux agent. Same bundler and fastlane versions.

**Comment by cpk1** [comment]
I'm running into the same issue on jenkins for any fastlane action :(
looking into it!

**Comment by cpk1** [comment]
Using the gemfile and adding `gem 'unicode-display_width', '1.3.0'` will get you back up and running again. I haven't had time yet to figure out why it's working locally but not on Jenkins if you use 1.3.2

@joshdholtz Not sure if it's worth Fastlane pinning the version or not until the root cause can be figured out?

This looks to be the commit that is causing this: https://github.com/janlelis/unicode-display_width/commit/679f789660308eeb908cd6f0b0aa1c65113537f9

**Comment by cpk1** [comment]
So not sure exactly what your set up looks like but I'll try to answer.
After editing your gemfile run "bundle update", this updates your gemfile.lock. You then commit both these files to version control.

If you've already done this and are still having trouble then probably the next step is to inspect your gemfile.lock file to see what is locking you at that specific version

**Comment by hjh5696** [comment]
Thanks for your quick reply! I already manually update gemfile.lock (change unicode-display_width version to 1.3.0) and then the error seems to be resolved. 
Is it a bug in unicode-display_width 1.3.2?
